src/material.js

```diff
@@ -66,6 +66,7 @@ export const defaultMaterialConfig = {
 export class Material {
     constructor(config) {
         this._json = lodash.cloneDeep(config || {});
+        this.setProp('isNonPeriodic', (config.tags || []).includes('non-periodic') || config.isNonPeriodic);
```

Comment:
moved the non-periodic property setting here from `bank_entity` in the web-app so that it only sets the value for materials rather than any bank_entity item.